models/demos/mnist/tests/test_perf_mnist.py

```diff
@@ -0,0 +1,127 @@
+from torch.utils.data import DataLoader
+from models.utility_functions import (
+    enable_persistent_kernel_cache,
+    disable_persistent_kernel_cache,
+)
+from models.perf.perf_utils import prep_perf_report
+from ttnn.model_preprocessing import preprocess_model_parameters
+from models.demos.mnist.reference.mnist import MnistModel
+from models.utility_functions import is_grayskull, is_wormhole_b0
+from models.perf.device_perf_utils import run_device_perf, check_device_perf, prep_device_perf_report
+from models.demos.mnist.tt import tt_mnist
+
+transform = transforms.Compose([transforms.ToTensor()])
+test_dataset = datasets.MNIST(root="./data", train=False, transform=None, download=True)
+
+
+def get_expected_times(tt_mnist):
+    if is_grayskull():
+        return {
+            tt_mnist: (3.54, 0.00905),
+        }[tt_mnist]
+    elif is_wormhole_b0():
+        return {
+            tt_mnist: (5.62, 0.0081),
+        }[tt_mnist]
+
+
+@pytest.mark.models_performance_bare_metal
+@pytest.mark.models_performance_virtual_machine
+@pytest.mark.parametrize(
+    "batch_size",
+    [128],
+)
+@pytest.mark.parametrize(
+    "tt_mnist",
+    [tt_mnist],
+)
+@pytest.mark.parametrize("device_params", [{"l1_small_size": 32768}], indirect=True)
+def test_performance_mnist(device, batch_size, tt_mnist, model_location_generator, reset_seeds):
+    state_dict = torch.load(model_location_generator("mnist_model.pt", model_subdir="mnist"))
+    model = MnistModel(state_dict)
+    model = model.eval()
+    disable_persistent_kernel_cache()
+    parameters = preprocess_model_parameters(
+        initialize_model=lambda: model,
+        convert_to_ttnn=lambda *_: True,
+        device=device,
+    )
+    transform = transforms.Compose([transforms.ToTensor()])
+    test_dataset = datasets.MNIST(root="./data", train=False, transform=transform, download=True)
+    dataloader = DataLoader(test_dataset, batch_size=batch_size)
+    x, labels = next(iter(dataloader))
+
+    test_input = ttnn.from_torch(x, dtype=ttnn.bfloat16, device=device)
+    durations = []
+    for _ in range(2):
+        start = time.time()
+
+        ttnn_output = tt_mnist.mnist(
+            device=device,
+            x=test_input,
+            batch_size=batch_size,
+            parameters=parameters,
+        )
+        end = time.time()
+        durations.append(end - start)
+
+    inference_and_compile_time, *inference_times = durations
+    average_inference_time = sum(inference_times) / len(inference_times)
+    expected_compile_time, expected_inference_time = get_expected_times(tt_mnist)
+
+    prep_perf_report(
+        model_name="MNIST",
+        batch_size=batch_size,
+        inference_and_compile_time=inference_and_compile_time,
+        inference_time=average_inference_time,
+        expected_compile_time=expected_compile_time,
+        expected_inference_time=expected_inference_time,
+        comments="",
+        inference_time_cpu=0.0,
+    )
+
```

Comment:
test should fail if perf is lower than expected

Reply:
Updated the file with the assert statement.

Reply:
the test ran in CI with 0.004, the expected inference time is 0.008, double of actual run. Even if the test slows down by 50% due to a bug, this test will not catch it. We probably need expected time to be closer to the actual run.